src/gpio_monitor.cpp

```diff
@@ -23,7 +27,30 @@ void GpioEventHandler::setEventHandlerForGpioPresence(
 void GpioMonitor::initHandlerForGpio(
     const std::shared_ptr<boost::asio::io_context>& i_ioContext)
 {
-    // ToDo Add implementation.
-    (void)i_ioContext;
+    try
+    {
+        std::vector<types::GpioPollingParameters>
+            l_pollingRequiredFrusParamsList =
+                jsonUtility::getListOfPollingParamsForFrus(m_sysCfgJsonObj);
+
+        if (!l_pollingRequiredFrusParamsList.empty())
```

Comment:
Don't need a check for empty. Just loop around, if there is no item, it will not enter the loop.

Reply:
Updated